front/pages/w/[wId]/u/gens/index.tsx

```diff
@@ -154,33 +154,49 @@ export class FunctionSingleArgStreamer {
   _curParsedPos: number;
   _arg: string;
   _handler: (token: string) => void;
+  _in_string: boolean;
 
   constructor(arg: string, handler: (token: string) => void) {
     this._arg = arg;
     this._textSoFar = "";
     this._curParsedPos = 0;
     this._handler = handler;
+    this._in_string = false;
   }
 
-  feed(token: string): void {
-    this._textSoFar += token;
-
-    let str = this._textSoFar + '"}';
-    if (this._textSoFar.trimEnd().endsWith('"')) {
+  _apply_processing(text: string): string {
+    let str = text + '"}';
+    if (text.trimEnd().endsWith('"')) {
       // If _textSoFar ends with a quote, we just add the } to the end.
-      str = this._textSoFar + "}";
+      str = text + "}";
     }
+    return str;
+  }
+
+  feed(token: string): void {
+    let currentTextSoFar = this._textSoFar + token;
+    let str = this._apply_processing(currentTextSoFar);
+    let obj;
 
     try {
-      const obj = JSON.parse(str);
+      obj = JSON.parse(str);
+    } catch (e) {
+      // trying to modify token to get it to work
+      const processedToken = token.replace(/\n/g, "\\n");
+      currentTextSoFar = this._textSoFar + processedToken;
+      str = this._apply_processing(currentTextSoFar);
+    }
+    try {
+      obj = JSON.parse(str);
+      this._textSoFar = currentTextSoFar;
```

Comment:
This should not be the messed up string. We want to keep the pure stream here and apply processing anytime we fail otherwise you risk to shout yourself in the foot deifnitively

Reply:
that line will only be reached if the parsing worked on the modification